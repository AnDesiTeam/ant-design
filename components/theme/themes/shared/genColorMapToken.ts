import type { MapToken } from '../../interface';
import type {
  BgPalettes,
  ErrorPalettes,
  InfoPalettes,
  PrimaryPalettes,
  SuccessPalettes,
  TextAlphaPalettes,
  WarningPalettes,
} from '../IPalettes';

interface PaletteSheets {
  primaryPalettes: PrimaryPalettes;
  successPalettes: SuccessPalettes;
  errorPalettes: ErrorPalettes;
  warningPalettes: WarningPalettes;
  infoPalettes: InfoPalettes;
  textAlphaPalettes: TextAlphaPalettes;
  bgPalettes: BgPalettes;
}

type ColorMapToken = Pick<
  MapToken,
  | 'colorPrimary'
  | 'colorPrimaryHover'
  | 'colorPrimaryActive'
  | 'colorPrimaryBorder'
  | 'colorPrimaryBorderHover'
  | 'colorSuccess'
  | 'colorSuccessBg'
  | 'colorSuccessBorder'
  | 'colorError'
  | 'colorErrorBg'
  | 'colorErrorBorder'
  | 'colorErrorHover'
  | 'colorErrorActive'
  | 'colorWarning'
  | 'colorWarningBg'
  | 'colorWarningBorder'
  | 'colorWarningHover'
  | 'colorWarningActive'
  | 'colorInfo'
  | 'colorInfoBg'
  | 'colorInfoBorder'
  | 'colorLink'
  | 'colorLinkHover'
  | 'colorLinkActive'
  | 'colorDefaultOutline'
  | 'colorBgLayout'
  | 'colorBgElevated'
  | 'colorBgContainer'
  | 'colorBgContainerSecondary'
  | 'colorBgContainerDisabled'
  | 'colorBgContent'
  | 'colorBgContentHover'
  | 'colorBorder'
  | 'colorBorderSecondary'
  | 'colorSplit'
  | 'colorText'
  | 'colorTextHeading'
  | 'colorTextSecondary'
  | 'colorTextDisabled'
  | 'colorTextPlaceholder'
  | 'colorAction'
  | 'colorActionHover'
  | 'colorHighlight'
>;

export default function genColorMapToken({
  primaryPalettes,
  successPalettes,
  errorPalettes,
  warningPalettes,
  infoPalettes,
  textAlphaPalettes,
  bgPalettes,
}: PaletteSheets): ColorMapToken {
  return {
    colorPrimary: primaryPalettes['6'],
    colorPrimaryHover: primaryPalettes['5'],
    colorPrimaryActive: primaryPalettes['7'],
    colorPrimaryBorder: primaryPalettes['3'],
    colorPrimaryBorderHover: primaryPalettes['4'],

    colorSuccess: successPalettes['6'],
    colorSuccessBg: successPalettes['1'],
    colorSuccessBorder: successPalettes['3'],

    colorError: errorPalettes['6'],
    colorErrorBg: errorPalettes['1'],
    colorErrorBorder: errorPalettes['3'],
    colorErrorHover: errorPalettes['5'],
    colorErrorActive: errorPalettes['7'],

    colorWarning: warningPalettes['6'],
    colorWarningBg: warningPalettes['1'],
    colorWarningBorder: warningPalettes['3'],
    colorWarningHover: warningPalettes['5'],
    colorWarningActive: warningPalettes['7'],

    colorInfo: infoPalettes['6'],
    colorInfoBg: infoPalettes['1'],
    colorInfoBorder: infoPalettes['3'],

    colorLink: primaryPalettes['6'],
    colorLinkHover: primaryPalettes['5'],
    colorLinkActive: primaryPalettes['7'],

    colorDefaultOutline: textAlphaPalettes['4'],

    // ============== Background ============== //
    colorBgLayout: bgPalettes['0'],
    colorBgElevated: bgPalettes['12'],
    colorBgContainer: bgPalettes['8'],
    colorBgContainerSecondary: textAlphaPalettes['4'],
    colorBgContainerDisabled: textAlphaPalettes['8'],
    colorBgContent: bgPalettes['15'],
    colorBgContentHover: bgPalettes['26'],

    // ============== Split ============== //
    colorBorder: bgPalettes['26'],
    colorBorderSecondary: bgPalettes['19'],
    colorSplit: textAlphaPalettes['12'],

    // ============== Text ============== //
    colorText: textAlphaPalettes['85'],
    colorTextHeading: textAlphaPalettes['85'],
    colorTextSecondary: textAlphaPalettes['45'],
    // TODO: 这个 30 估计要改成 25
    colorTextDisabled: textAlphaPalettes['30'],
    colorTextPlaceholder: textAlphaPalettes['25'],

    colorAction: textAlphaPalettes['45'],
    colorActionHover: textAlphaPalettes['85'],

    colorHighlight: errorPalettes['5'],
  };
}
