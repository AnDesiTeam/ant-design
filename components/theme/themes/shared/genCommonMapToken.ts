import genFontMapToken from 'antd/es/theme/themes/shared/genFontMapToken';
import type { CommonMapToken, SeedToken } from '../../interface';
import genRadius from './genRadius';

export default function genCommonMapToken(token: SeedToken): CommonMapToken {
  const { motionUnit, motionBase, borderRadius, lineWidth, fontSize } = token;

  return {
    // motion
    motionDurationFast: `${(motionBase + motionUnit).toFixed(1)}s`,
    motionDurationMid: `${(motionBase + motionUnit * 2).toFixed(1)}s`,
    motionDurationSlow: `${(motionBase + motionUnit * 3).toFixed(1)}s`,

    // font
    ...genFontMapToken(fontSize),

    // line
    lineWidthBold: lineWidth + 1,

    // radius
    ...genRadius(borderRadius),
  };
}
