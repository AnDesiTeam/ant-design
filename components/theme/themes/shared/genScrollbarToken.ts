import type { ScrollBarToken } from '../../interface/maps';

function genScrollbarToken(): ScrollBarToken {
  return {
    scrollbarColor: 'auto',
    scrollbarWidth: 'auto',
    scrollbarGutter: 'auto',
  };
}

export default genScrollbarToken;
