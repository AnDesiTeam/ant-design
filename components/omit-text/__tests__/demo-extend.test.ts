import { extendTest } from '../../../tests/shared/demoTest';

extendTest('omit-text');
