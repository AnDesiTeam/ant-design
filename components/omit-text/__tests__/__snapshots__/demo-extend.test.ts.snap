// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ./components/omit-text/demo/basic.md extend context correctly 1`] = `
Array [
  <div
    class="ant-omit-text"
    style="max-width:120px"
  >
    这是一段很长很长很长很长很长很长很长很长很长很长很长很长很长很长很长很长的文本
  </div>,
  <div>
    <div
      class="ant-tooltip"
      style="opacity:0"
    >
      <div
        class="ant-tooltip-content"
      >
        <div
          class="ant-tooltip-arrow"
        >
          <span
            class="ant-tooltip-arrow-content"
          />
        </div>
        <div
          class="ant-tooltip-inner"
          role="tooltip"
        >
          这是一段很长很长很长很长很长很长很长很长很长很长很长很长很长很长很长很长的文本
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`renders ./components/omit-text/demo/type.md extend context correctly 1`] = `
Array [
  <div
    class="ant-omit-text"
    style="max-width:120px"
  >
    这是一段很长很长很长很长很长很长很长很长很长很长很长很长很长很长很长很长的文本
  </div>,
  <div>
    <div
      class="ant-popover ant-popconfirm"
      style="opacity:0"
    >
      <div
        class="ant-popover-content"
      >
        <div
          class="ant-popover-arrow"
        >
          <span
            class="ant-popover-arrow-content"
          />
        </div>
        <div
          class="ant-popover-inner"
          role="tooltip"
        >
          <div
            class="ant-popover-inner-content"
          >
            <div
              class="ant-popover-message"
            >
              <span
                aria-label="exclamation-circle"
                class="anticon anticon-exclamation-circle"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="exclamation-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
                  />
                </svg>
              </span>
              <div
                class="ant-popover-message-title"
              >
                这是一段很长很长很长很长很长很长很长很长很长很长很长很长很长很长很长很长的文本
              </div>
            </div>
            <div
              class="ant-popover-buttons"
            >
              <button
                class="ant-btn ant-btn-default ant-btn-sm"
                type="button"
              >
                <span>
                  Cancel
                </span>
              </button>
              <button
                class="ant-btn ant-btn-primary ant-btn-sm"
                type="button"
              >
                <span>
                  OK
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`renders ./components/omit-text/demo/width.md extend context correctly 1`] = `
Array [
  <div
    class="ant-omit-text"
    style="max-width:200px"
  >
    这是一段很长很长很长很长很长很长很长很长很长很长很长很长很长很长很长很长的文本
  </div>,
  <div>
    <div
      class="ant-tooltip"
      style="opacity:0"
    >
      <div
        class="ant-tooltip-content"
      >
        <div
          class="ant-tooltip-arrow"
        >
          <span
            class="ant-tooltip-arrow-content"
          />
        </div>
        <div
          class="ant-tooltip-inner"
          role="tooltip"
        >
          这是一段很长很长很长很长很长很长很长很长很长很长很长很长很长很长很长很长的文本
        </div>
      </div>
    </div>
  </div>,
]
`;
